src/types/ScreenInfoUiaProvider.cpp

```diff
@@ -31,9 +31,22 @@ SAFEARRAY* BuildIntSafeArray(_In_reads_(length) const int* const data, const int
     return psa;
 }
 
+ScreenInfoUiaProvider::ScreenInfoUiaProvider(_In_ Microsoft::Console::Render::IRenderData* pData,
+                                             _In_ WindowUiaProviderBase* const pUiaParent,
+                                             _In_ std::function<RECT(void)> GetBoundingRect) :
+    _pUiaParent(pUiaParent),

```

Comment:
actually a pointer already _is_ an optional... i'm not sure this gets us much except an automatic exception on deref. but then, we'd gett hat on null deref anyway.

Reply:
I think it's better to detect what you can on construction than let a null deref happen later...